backend/cmd/cluster.go

```diff
@@ -41,12 +67,36 @@ func GetClustersFromKubeConfigFile(kubeConfigPath string) ([]Cluster, error) {
 	return clusters, nil
 }
 
+func GetClusterAuthConfig(clusterName string) (*AuthConfig, error) {
+	if authConfig, ok := authConfigs[clusterName]; ok {
+		return authConfig, nil
+	}
+	return nil, fmt.Errorf("Couln't find auth config for cluster %s", clusterName)
+}
+
 func GetOwnCluster() (*Cluster, error) {
 	config, err := rest.InClusterConfig()
 	if err != nil {
 		return nil, err
 	}
 
+	var authConfig AuthConfig
+	authProvider := config.AuthProvider
+	if authProvider != nil {
+		authType := authProvider.Name
+		clientID := authProvider.Config["client-id"]
+		clientSecret := authProvider.Config["client-secret"]
+		idpIssuerURL := authProvider.Config["idp-issuer-url"]
+
+		authConfig.ClientID = clientID
+		authConfig.ClientSecret = clientSecret
+		authConfig.IdpIssuerURL = idpIssuerURL
+		authConfig.Type = authType
+	} else {
+		authConfig.Type = "token"
+	}
+	authConfigs[OWN_CLUSTER_NAME] = &authConfig
+
```

Comment:
I don't think the in-cluster config will have the auth provider set up like this.
Instead, we should access the related config as arguments to the server (`--oidc-client-id`, `--oidc-secret`, ...).